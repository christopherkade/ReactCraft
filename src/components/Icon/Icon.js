import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Wrapper = styled.div`
  display: inline-flex;
  border: 0.125rem ridge #696967;
  border-radius: 0.25rem;
`

const StyledIcon = styled.i`
  display: inline-block;
  background-image: url('${({ name }) => name}');;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  width: ${({ width }) => width}px;
  height: ${({ height }) => height}px;
  border: 1px solid #CACBCA;
  border-radius: 3px;
`;

const Icon = ({ name, width, height }) => (
  <Wrapper width={width} height={height}>
    <StyledIcon name={name} width={width} height={height} />
  </Wrapper>
);

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
};

Icon.defaultProps = {
  width: 32,
  height: 32,
};

export default Icon
