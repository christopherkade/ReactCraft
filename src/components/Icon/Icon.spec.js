import React from 'react';
import Icon from './Icon';
import { render } from 'react-testing-library';

describe('<Icon />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Icon name="mockedName"/>);
      expect(container).toMatchSnapshot();
    });
  });
});
