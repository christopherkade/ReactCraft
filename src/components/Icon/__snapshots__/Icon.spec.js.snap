// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa couKLk"
    height="32"
    width="32"
  >
    <i
      class="sc-bwzfXH GmSjy"
      height="32"
      name="mockedName"
      width="32"
    />
  </div>
</div>
`;
